src/Dialect/Mlir/IndexExpr.cpp

```diff
@@ -1165,6 +1165,12 @@ QuestionmarkIndexExpr::QuestionmarkIndexExpr() : IndexExpr() {
   indexExprObj->initAsQuestionmark();
 }
 
+QuestionmarkIndexExpr::QuestionmarkIndexExpr(Value val, int64_t axis)
```

Comment:
I think the val here is of type Tensor or Memref, for this to work, correct? Maybe you could rename `val` to be bit more explicit. Maybe `tensorOrMemRefVal` or `arrayVal` and `axis` becoming `index`. It will help comprehension. 

Please change in other similar instances below